lib/https.js

```diff
@@ -572,30 +572,34 @@ Agent.prototype._cacheSession = function _cacheSession(key, session) {
   if (this.maxCachedSessions === 0)
     return;
 
+  // Cache property accesses for better performance
+  const { map, list } = this._sessionCache;
```

Comment:
I would avoid destructuring and just extract `map` here, and then `list` just before its used.